### DISCUSSION TITLE
PHP PDO MySQL Class port to Deno?

### DISCUSSION THREAD

**Discussion by suchislife801** [denoland/deno#9229] [Ideas]
Greetings.

Currently, the only reason I still use PHP is for the superb PDO Class it has, used for MySQL connections and data.

It would be fantastic if Deno supported these right out of the box.

3rd party libraries exist but aren't really usable for MySQL or user friendly. And not even close to PDO.

I'm a PHP developer and since PHP is open source I was wondering if you could port PDO to Deno. This would greatly increase Deno popularity and the code is technically already written; PHP is open source.

The real PDO benefits are:

- security (usable prepared statements)
- usability (many helper functions to automate routine operations)
- re-usability (unified API to access multitude of databases, from SQLite to Oracle)

Ideally only the MySQL portion of the PDO Class would be ported at first to greatly reduce the time it takes to implement.

[https://www.php.net/manual/en/book.pdo.php](https://www.php.net/manual/en/book.pdo.php)

[https://www.php.net/downloads](https://www.php.net/downloads)

**Comment by vicary** [comment]
https://github.com/denolib/typeorm is the closest thing you get right now.

**Comment by minecrawler** [comment]
I don't think it's a good idea to add a MySQL connector to Deno, as that would bloat the core. Also, then there should also be support for other 3rd Party applications, like MongoDB, RabbitMQ, etc., which will all only be used by a small subset of Deno users.

Which is why I recommend starting a separate project to port PDO to TypeScript and base it on Deno; or even better: send a PR to a good existing MySQL library for NodeJS which adds optional support for Deno as a runtime environment.

**Reply by felix-schindler** [reply]
Why not add it to Deno Standard Modules then?
